lib/SplatArgPass.cpp

```diff
@@ -41,184 +52,142 @@ namespace clspv {
-    return "_Z3maxDv2_DhS_";
-  } else if (Name.equals("_Z3maxDv3_DhDh")) {
-    return "_Z3maxDv3_DhS_";
-  } else if (Name.equals("_Z3maxDv4_DhDh")) {
-    return "_Z3maxDv4_DhS_";
-  } else if (Name.equals("_Z4fmaxDv2_ff")) {
-    return "_Z4fmaxDv2_fS_";
-  } else if (Name.equals("_Z4fmaxDv3_ff")) {
-    return "_Z4fmaxDv3_fS_";
-  } else if (Name.equals("_Z4fmaxDv4_ff")) {
-    return "_Z4fmaxDv4_fS_";
-  } else if (Name.equals("_Z4fmaxDv2_DhDh")) {
-    return "_Z4fmaxDv2_DhS_";
-  } else if (Name.equals("_Z4fmaxDv3_DhDh")) {
-    return "_Z4fmaxDv3_DhS_";
-  } else if (Name.equals("_Z4fmaxDv4_DhDh")) {
-    return "_Z4fmaxDv4_DhS_";
-
-  } else if (Name.equals("_Z3minDv2_ii")) {
-    return "_Z3minDv2_iS_";
-  } else if (Name.equals("_Z3minDv3_ii")) {
-    return "_Z3minDv3_iS_";
-  } else if (Name.equals("_Z3minDv4_ii")) {
-    return "_Z3minDv4_iS_";
-  } else if (Name.equals("_Z3minDv2_jj")) {
-    return "_Z3minDv2_jS_";
-  } else if (Name.equals("_Z3minDv3_jj")) {
-    return "_Z3minDv3_jS_";
-  } else if (Name.equals("_Z3minDv4_jj")) {
-    return "_Z3minDv4_jS_";
-  } else if (Name.equals("_Z3minDv2_ff")) {
-    return "_Z3minDv2_fS_";
-  } else if (Name.equals("_Z3minDv3_ff")) {
-    return "_Z3minDv3_fS_";
-  } else if (Name.equals("_Z3minDv4_ff")) {
-    return "_Z3minDv4_fS_";
-  } else if (Name.equals("_Z3minDv2_DhDh")) {
-    return "_Z3minDv2_DhS_";
-  } else if (Name.equals("_Z3minDv3_DhDh")) {
-    return "_Z3minDv3_DhS_";
-  } else if (Name.equals("_Z3minDv4_DhDh")) {
-    return "_Z3minDv4_DhS_";
-  } else if (Name.equals("_Z4fminDv2_ff")) {
-    return "_Z4fminDv2_fS_";
-  } else if (Name.equals("_Z4fminDv3_ff")) {
-    return "_Z4fminDv3_fS_";
-  } else if (Name.equals("_Z4fminDv4_ff")) {
-    return "_Z4fminDv4_fS_";
-  } else if (Name.equals("_Z4fminDv2_DhDh")) {
-    return "_Z4fminDv2_DhS_";
-  } else if (Name.equals("_Z4fminDv3_DhDh")) {
-    return "_Z4fminDv3_DhS_";
-  } else if (Name.equals("_Z4fminDv4_DhDh")) {
-    return "_Z4fminDv4_DhS_";
-
-  } else if (Name.equals("_Z3mixDv2_fS_f")) {
-    return "_Z3mixDv2_fS_S_";
-  } else if (Name.equals("_Z3mixDv3_fS_f")) {
-    return "_Z3mixDv3_fS_S_";
-  } else if (Name.equals("_Z3mixDv4_fS_f")) {
-    return "_Z3mixDv4_fS_S_";
-  } else if (Name.equals("_Z3mixDv2_DhS_Dh")) {
-    return "_Z3mixDv2_DhS_S_";
-  } else if (Name.equals("_Z3mixDv3_DhS_Dh")) {
-    return "_Z3mixDv3_DhS_S_";
-  } else if (Name.equals("_Z3mixDv4_DhS_Dh")) {
-    return "_Z3mixDv4_DhS_S_";
+namespace {
+int log2i(int val) {
+  if (val <= 0)
+    return -1;
+  int log2 = 0;
+  while (val >>= 1)
+    log2++;
+  return log2;
+}
+} // namespace
+
+// Programmatically convert mangled_name to vectorized version
+std::string
+SplatArgPass::getSplatName(const Builtins::FunctionInfo &func_info,
+                           const Builtins::ParamTypeInfo &param_info,
+                           bool three_params) {
+  const char *type_code = "f";
+  int index = log2i(param_info.byte_len);
```

Comment:
Since byte_len must be positive, I'd suggest dropping the log2i implementation and using MathExtras.h's Log2_32.